mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -528,7 +531,10 @@ public void deleteMailbox(final MailboxPath mailboxPath, final MailboxSession se
         assertIsOwner(session, mailboxPath);
         final MailboxMapper mapper = mailboxSessionMapperFactory.getMailboxMapper(session);
 
-        Mailbox mailbox = mapper.execute((Mapper.Transaction<Mailbox>) () -> {
+        QuotaRoot quotaRoot = quotaRootResolver.getQuotaRoot(mailboxPath);
```

Comment:
these computations should be done in the transaction to benefits from snapshot isolation when using JPA.